File: JavaScript/countBits.js
/*
|--------------------------------------------------------------------------
| Bit Counting
|--------------------------------------------------------------------------
|
*/

function countBits(n) {
   return n.toString(2).split('').filter(i => i == '1').length;
}


function countBits(n) {
   for (c = 0; n; n >>= 1) c += n & 1
   return c;
}


var countBits = function (n) {
   return n.toString(2).replace(/0/g, '').length;
}


var countBits = function (n) {
   a = n.toString(2).match(/1/g);
   return a == null ? 0 : a.length;
}


var countBits = function (n) {
   return n.toString(2).split("").reduce((a, b) => parseInt(a) + parseInt(b), 0);
}


console.log(countBits(0), 0);
console.log(countBits(4), 1);
console.log(countBits(7), 3);
console.log(countBits(9), 2);
console.log(countBits(10), 2);
console.log(countBits(1234), 5);

File: Python/6py 1/count_bits.py
"""
|--------------------------------------------------------------------------
| Bit Counting
|--------------------------------------------------------------------------
|
"""

def count_bits(n):
   return len(list(filter(lambda i: i == '1' , bin(n)[2:])))


def countBits(n):
   return bin(n).count("1")


def countBits(n):
   res = 0
   while n > 0:
      res += n % 2
      n >>= 1
   return res


def countBits(n):
   return '{:b}'.format(n).count('1')


def countBits(n):
   return (n & 1) + countBits(n >> 1) if n else 0


print(count_bits(0), 0)
print(count_bits(4), 1)
print(count_bits(7), 3)
print(count_bits(9), 2)
print(count_bits(10), 2)